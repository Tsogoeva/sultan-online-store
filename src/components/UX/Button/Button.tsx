import { FC } from "react";
import styles from './button.module.scss';

interface IButtonProps {
	text: string;
	icon: string;
}

const Button: FC<IButtonProps> = ({ text, icon }) => {
	return (
		<div className={styles.container}>
			<button type="button">
				{text}
				<img src={icon} alt="Иконка" />
			</button>
		</div>
	)
}

export default Button;