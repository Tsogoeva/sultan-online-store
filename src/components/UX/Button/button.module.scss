.container {
	.big_size {
		width: 195px;
		height: 60px;
	}

	.mini_size {
		width: 153px;
		height: 45px;
	}

	button {
		width: 195px;
		height: 60px;

		background-color: rgba(255, 200, 94, 1);
		border-radius: 30px;

		color: rgba(255, 255, 255, 1);
		font-weight: 700;
		line-height: 17px;

		img {
			margin: 2px 0 0 12px;
		}
	}
}
