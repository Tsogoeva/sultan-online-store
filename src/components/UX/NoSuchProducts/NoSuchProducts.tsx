import { FC } from 'react';
import styles from './no-such-products.module.scss';


const NoSuchProducts: FC = () => {
	return (
		<div className={styles.not_found_container}>
			<h4
				className={styles.not_found_info}
			>
				Товары по заданным параметрам отсутствуют!
			</h4>
		</div>
	)
}

export default NoSuchProducts;
