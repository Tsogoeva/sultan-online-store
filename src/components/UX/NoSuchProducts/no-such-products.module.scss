.not_found_container {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;

	.not_found_info {
		margin: 5px;
		padding: 10px 8px;
		text-align: center;
		font-size: 20px;

		color: rgba(63, 78, 101, 1);
	}
}
