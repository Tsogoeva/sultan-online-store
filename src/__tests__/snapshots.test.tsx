import renderer from 'react-test-renderer';
import Button from '../components/UX/Button/Button';
import NoSuchProducts from '../components/UX/NoSuchProducts/NoSuchProducts';
import cartIcon from '../components/UX/ButtonToCartFromProduct/assets/mini-cart-icon.svg';

describe('Snapshots renders', () => {

	test('render NoSuchProducts component', () => {
		const tree = renderer
			.create(<NoSuchProducts />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('render Button component', () => {
		const tree = renderer
			.create(<Button text={'Тест'} icon={cartIcon} size={'big'} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
})
