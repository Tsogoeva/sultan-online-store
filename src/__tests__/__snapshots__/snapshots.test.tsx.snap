// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots renders render Button component 1`] = `
<div
  className="container"
>
  <button
    className="big_size"
    type="button"
  >
    Тест
    <img
      alt="Иконка"
      src="mini-cart-icon.svg"
    />
  </button>
</div>
`;

exports[`Snapshots renders render NoSuchProducts component 1`] = `
<div
  className="not_found_container"
>
  <h4
    className="not_found_info"
  >
    Товары по заданным параметрам отсутствуют!
  </h4>
</div>
`;
